diff -urN --no-dereference orig/rules/base.lst new/rules/base.lst
--- orig/rules/base.lst	2021-05-26 21:10:11.010635562 +0200
+++ new/rules/base.lst	2021-05-26 21:10:11.042635258 +0200
@@ -678,6 +678,7 @@
   legacy          ch: German (Switzerland, legacy)
   de_nodeadkeys   ch: German (Switzerland, no dead keys)
   de_sundeadkeys  ch: German (Switzerland, with Sun dead keys)
+  dvorak          ch: German (Switzerland, Dvorak)
   fr              ch: French (Switzerland)
   fr_nodeadkeys   ch: French (Switzerland, no dead keys)
   fr_sundeadkeys  ch: French (Switzerland, with Sun dead keys)
diff -urN --no-dereference orig/rules/base.xml new/rules/base.xml
--- orig/rules/base.xml	2021-05-26 21:10:11.010635562 +0200
+++ new/rules/base.xml	2021-05-26 21:10:11.042635258 +0200
@@ -5164,6 +5164,14 @@
         </variant>
         <variant>
           <configItem>
+            <name>dvorak</name>
+            
+            <shortDescription>de</shortDescription>
+            <description>German (Switzerland, Dvorak)</description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>fr</name>
             <!-- Keyboard indicator for French layouts -->
             <shortDescription>fr</shortDescription>
diff -urN --no-dereference orig/rules/evdev.lst new/rules/evdev.lst
--- orig/rules/evdev.lst	2021-05-26 21:10:11.010635562 +0200
+++ new/rules/evdev.lst	2021-05-26 21:10:11.042635258 +0200
@@ -678,6 +678,7 @@
   legacy          ch: German (Switzerland, legacy)
   de_nodeadkeys   ch: German (Switzerland, no dead keys)
   de_sundeadkeys  ch: German (Switzerland, with Sun dead keys)
+  dvorak          ch: German (Switzerland, Dvorak)
   fr              ch: French (Switzerland)
   fr_nodeadkeys   ch: French (Switzerland, no dead keys)
   fr_sundeadkeys  ch: French (Switzerland, with Sun dead keys)
diff -urN --no-dereference orig/rules/evdev.xml new/rules/evdev.xml
--- orig/rules/evdev.xml	2021-05-26 21:10:11.010635562 +0200
+++ new/rules/evdev.xml	2021-05-26 21:10:11.042635258 +0200
@@ -5164,6 +5164,14 @@
         </variant>
         <variant>
           <configItem>
+            <name>dvorak</name>
+            
+            <shortDescription>de</shortDescription>
+            <description>German (Switzerland, Dvorak)</description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>fr</name>
             <!-- Keyboard indicator for French layouts -->
             <shortDescription>fr</shortDescription>
diff -urN --no-dereference orig/symbols/ch new/symbols/ch
--- orig/symbols/ch	2021-05-26 21:10:11.014635524 +0200
+++ new/symbols/ch	2021-05-26 21:10:11.042635258 +0200
@@ -76,6 +76,54 @@
 };
 
 partial alphanumeric_keys
+xkb_symbols "dvorak" {
+    // modify the basic Swiss German layout not to have any dead keys
+    include "ch(de_sundeadkeys)"
+    name[Group1]= "German (Switzerland, dvorak)";
+
+    key <TLDE> { [    section,     degree                             ] };
+    key <AD01>	{ [udiaeresis,     egrave,           at,  Greek_OMEGA ] };
+    key <AD02>	{ [     comma,  semicolon                             ] };
+    key <AD03>	{ [    period,      colon                             ] };
+    key <AD04>	{ [         p,          P,        thorn,        THORN ] };
+    key <AD05>	{ [         y,          Y                             ] };
+    key <AD06>  { [         f,          F,      dstroke,  ordfeminine ] };
+    key <AD07>  { [         g,          G,          eng,          ENG ] };
+    key <AD08>  { [         c,          C,         cent,    copyright ] };
+    key <AD09>  { [         t,          T,       tslash,       Tslash ] };
+    key <AD10>  { [         z,          Z                             ] };
+    key <AD11> { [       less,    greater,  bracketleft               ] };
+    key <AD12>	{ [ dead_diaeresis, exclam, bracketright              ] };
+
+    key <AC01>	{ [         a,          A,           ae,           AE ] };
+    key <AC02>	{ [         o,          O,       oslash,     Ooblique ] };
+    key <AC03>	{ [         e,          E,     EuroSign               ] };
+    key <AC04>	{ [         i,          I,   rightarrow,     idotless ] };
+    key <AC05>	{ [         u,          U,    downarrow,      uparrow ] };
+    key <AC06>	{ [         h,          H,      hstroke,      Hstroke ] };
+    key <AC07>	{ [         d,          D,          eth,          ETH ] };
+    key <AC08>	{ [         r,          R,    paragraph,   registered ] };
+    key <AC09>	{ [         n,          N,            n,            N ] };
+    key <AC10>	{ [         s,          S,       ssharp,      section ] };
+    key <AC11>	{ [         l,          L,    braceleft               ] };
+    key <BKSL>	{ [     dollar,   sterling,   braceright              ] };
+
+    key <AB01>  { [ odiaeresis,     eacute                            ] };
+    key <AB02>  { [         q,          Q,           at,  Greek_OMEGA ] };
+    key <AB03>  { [         j,          J,    dead_hook,    dead_horn ] };
+    key <AB04>  { [         k,          K,          kra,    ampersand ] };
+    key <AB05>  { [         x,          X, guillemotright,    greater ] };
+    key <AB06>  { [         b,          B, rightdoublequotemark, rightsinglequotemark ] };
+    key <AB07>  { [         m,          M,           mu,    masculine ] };
+    key <AB08>  { [         w,          W,      lstroke,      Lstroke ] };
+    key <AB09>  { [         v,          V, leftdoublequotemark, leftsinglequotemark ]   };
+    key <AB10>  { [     minus, underscore                            ] };
+
+
+    key <LSGT>  { [ adiaeresis,     agrave,    backslash             ] };
+};
+
+partial alphanumeric_keys 
 xkb_symbols "fr" {
 
     include "ch(basic)"
